python/tank_vendor/shotgun_authentication/errors.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+All custom exceptions that this module emits are defined here.
+"""
+
+
+class ShotgunAuthenticationError(Exception):
+    """
+    Base class for all exceptions coming out from this module.
+    """
+    pass
+
+
+class AuthenticationError(ShotgunAuthenticationError):
+    """
+    Exception that indicates an authentication error has occurred.
+    """
+    pass
+
+
+class InvalidCredentials(ShotgunAuthenticationError):
```

Comment:
Maybe a stupid/irrelevant question, but just double checking: Should `InvalidCredentials` derive from `ShotgunAuthenticationError`. I am just thinking that it sounds to me that *invalid credentials* is a type of *authentication error*. Does that make sense?